### PR TITLE
Support React Hooks in examples

### PR DESCRIPTION
**Pull request by eragon512**
Implements the proposed solution for #1343

### CODE REVIEW

File: src/client/rsg-components/ReactExample/ReactExample.js [hunk 1]
```diff
@@ -6,6 +6,13 @@ import splitExampleCode from '../../utils/splitExampleCode';
 
 /* eslint-disable react/no-multi-comp */
 
+function FunctionComponentWrapper(props) {
```
Comment (sapegin) [inline src/client/rsg-components/ReactExample/ReactExample.js]: Could you add a comment here, explaining that we need it to support hooks in examples?

Comment (eragon512) [reply src/client/rsg-components/ReactExample/ReactExample.js]: added the comment, as requested

**Comment by eragon512** [comment]
~~~~
Could you please add some tests and replace initialState / setState in some examples to show how it works?
~~~~

@sapegin I added an extra example, because we still support the old state method, right?

**Comment by sapegin** [comment]
Awesome, thanks!

**Pull request merged**